### DISCUSSION TITLE
Option to pass LiquidCrystal object as a pointer in setup function

### DISCUSSION THREAD

**Discussion by dejanmeznarc** [forntoh/LcdMenu#94] [Ideas]
I'm trying to integrate this library into existing project that uses LCD to show some information. I'd like to use this library only as settings menu.

It would be really useful if I could do a custom LCD setup and have ability to use/edit LCD in other parts of program.

```c++
...
LcdMenu menu(LCD_ROWS, LCD_COLS);
LiquidCrystal lcd(rs, en, d4, d5, d6, d7);

void setup() {
    ...
    lcd.begin(16,4);
    ...
    menu.setup(&lcd, mainMenu); // passing my own lcd object to the library
}
...

```

**Comment by forntoh** [comment]
I'm already exploring decoupling the menu display from the menu control.
It will allow anyone to use any display method they want while keeping the core menu-controlling features this library has.

Ther's still a long way to go, I work on it only when I get spare time

**Reply by dejanmeznarc** [reply]
That is even better.  

I've already implemented this for myself. Should I open up a pr, if anybody needs the feature in the meantime?

**Comment by forntoh** [comment]
Added in **v5.0.0**
